### PR TITLE
Follower model and service

### PR DESCRIPTION
**Pull request by dxahtepb**
Implemented service for managing which users are associated with a feedback as a followers. For now there is two types of followage: as an assignee or as a watcher. This service may be used to send a notifiaction to watchers on new comments, for instance.

### CODE REVIEW

**Review by NeHo4y — APPROVED** [review]

File: server/src/main/kotlin/com/github/neho4y/follower/domain/FeedbackFollower.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    val followerType: FeedbackFollowerType,
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    val id: Long = 0
+)
+
+enum class FeedbackFollowerType {
+    ASSIGNEE,
+    WATCHER
```
Comment (maxxkut) [inline server/src/main/kotlin/com/github/neho4y/follower/domain/FeedbackFollower.kt:25]: u can add followerType : AUTHOR

File: server/src/main/kotlin/com/github/neho4y/follower/domain/repository/FeedbackFollowerRepository.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.github.neho4y.follower.domain.FeedbackFollower
+import com.github.neho4y.follower.domain.FeedbackFollowerType
+import org.springframework.data.jpa.repository.JpaRepository
+
+interface FeedbackFollowerRepository : JpaRepository<FeedbackFollower, Long> {
+    fun getAllByUserId(userId: Long): List<FeedbackFollower>
+    fun getAllByUserIdAndFollowerType(userId: Long, type: FeedbackFollowerType): List<FeedbackFollower>
+    fun getAllByFeedbackId(feedbackId: Long): List<FeedbackFollower>
+    fun getAllByFeedbackIdAndFollowerType(feedbackId: Long, type: FeedbackFollowerType): List<FeedbackFollower>
```
Comment (maxxkut) [inline server/src/main/kotlin/com/github/neho4y/follower/domain/repository/FeedbackFollowerRepository.kt]: u can use Filter for example JPA specification, but current realization is ok

File: server/src/main/kotlin/com/github/neho4y/follower/service/FeedbackFollowerService.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.github.neho4y.follower.service
+
+import com.github.neho4y.feedback.domain.Feedback
+import com.github.neho4y.follower.domain.FeedbackFollower
+import com.github.neho4y.follower.domain.FeedbackFollowerType
+import com.github.neho4y.user.domain.User
+
+interface FeedbackFollowerService {
+    fun getAllFollowsForUser(user: User, followerType: FeedbackFollowerType? = null): List<FeedbackFollower>
```
Comment (maxxkut) [inline server/src/main/kotlin/com/github/neho4y/follower/service/FeedbackFollowerService.kt]: same as upper comment

**Review by maxxkut — CHANGES_REQUESTED** [review]

File: server/src/main/kotlin/com/github/neho4y/follower/domain/FeedbackFollower.kt (hunk 1 above)
Comment (NeHo4y) [reply server/src/main/kotlin/com/github/neho4y/follower/domain/FeedbackFollower.kt:25]: Also consider single author validation

Comment (dxahtepb) [reply server/src/main/kotlin/com/github/neho4y/follower/domain/FeedbackFollower.kt:25]: We already have author field in the Feedback object. Followers supposed to be used for notifications. And author will be a watcher, so one may unsubscribe from own feedback.

**Comment by dxahtepb** [comment]
Will not merge develop in this branch before #8 is merged

**Review by NeHo4y — APPROVED** [review]

**Review by maxxkut — APPROVED** [review]

File: server/src/main/kotlin/com/github/neho4y/follower/domain/repository/FeedbackFollowerSpecificationRepository.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.github.neho4y.follower.domain.repository
+
+import com.github.neho4y.follower.domain.FeedbackFollower
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+
+interface FeedbackFollowerSpecificationRepository : JpaRepository<FeedbackFollower, Long>,
```
Comment (Walingar) [inline server/src/main/kotlin/com/github/neho4y/follower/domain/repository/FeedbackFollowerSpecificationRepository.kt:7]: `JpaRepository<FeedbackFollower, Long>` тогда тоже лучше на новую строчку вынести

File: server/src/main/kotlin/com/github/neho4y/follower/service/FeedbackFollowerService.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.github.neho4y.follower.service
+
+import com.github.neho4y.follower.domain.FeedbackFollower
+import com.github.neho4y.follower.model.FollowerDto
+import com.github.neho4y.follower.model.FollowerFilterDto
+
+interface FeedbackFollowerService {
```
Comment (Walingar) [inline server/src/main/kotlin/com/github/neho4y/follower/service/FeedbackFollowerService.kt:7]: Аналогично с другим сервисом: лучше делать методы саспендящимися или возвращать Future

**Pull request merged**